/*
 * Copyright 2006 Sun Microsystems, Inc.  All rights reserved.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the
 * "Software"), to deal in the Software without restriction, including
 * without limitation the rights to use, copy, modify, merge, publish,
 * distribute, and/or sell copies of the Software, and to permit persons
 * to whom the Software is furnished to do so, provided that the above
 * copyright notice(s) and this permission notice appear in all copies of
 * the Software and that both the above copyright notice(s) and this
 * permission notice appear in supporting documentation.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
 * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
 * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
 * OF THIRD PARTY RIGHTS. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
 * HOLDERS INCLUDED IN THIS NOTICE BE LIABLE FOR ANY CLAIM, OR ANY SPECIAL
 * INDIRECT OR CONSEQUENTIAL DAMAGES, OR ANY DAMAGES WHATSOEVER RESULTING
 * FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
 * NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
 * WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 *
 * Except as contained in this notice, the name of a copyright holder
 * shall not be used in advertising or otherwise to promote the sale, use
 * or other dealings in this Software without prior written authorization
 * of the copyright holder.
 *
 */

Changes made before the source was checked into Solaris SCCS/Teamware 
repositories that I couldn't find comments for nor guess why they were made.

diff -urp -x '*~' driver/demo-Gtk.c driver/demo-Gtk.c
--- driver/demo-Gtk.c	2006-05-10 08:49:43.010409000 -0700
+++ driver/demo-Gtk.c	2006-05-10 08:49:50.661964000 -0700
@@ -601,12 +601,13 @@ exit_menu_cb (GtkMenuItem *menuitem, gpo
   gtk_main_quit ();
 }
 
-static void
+static gboolean
 wm_toplevel_close_cb (GtkWidget *widget, GdkEvent *event, gpointer data)
 {
   state *s = (state *) data;
   flush_dialog_changes_and_save (s);
   gtk_main_quit ();
+  return TRUE;
 }
 
 
@@ -922,7 +923,7 @@ manual_cb (GtkButton *button, gpointer u
 {
   state *s = global_state_kludge;  /* I hate C so much... */
   saver_preferences *p = &s->prefs;
-  GtkList *list_widget = GTK_LIST (name_to_widget (s, "list"));
+  GtkWidget *list_widget = name_to_widget (s, "list");
   int list_elt = selected_list_element (s);
   int hack_number;
   char *name, *name2, *cmd, *str;
@@ -930,7 +931,7 @@ manual_cb (GtkButton *button, gpointer u
   hack_number = s->list_elt_to_hack_number[list_elt];
 
   flush_dialog_changes_and_save (s);
-  ensure_selected_item_visible (GTK_WIDGET (list_widget));
+  ensure_selected_item_visible (list_widget);
 
   name = strdup (p->screenhacks[hack_number]->command);
   name2 = name;
@@ -1974,11 +1975,12 @@ settings_ok_cb (GtkButton *button, gpoin
   gtk_widget_hide (s->popup_widget);
 }
 
-static void
+static gboolean
 wm_popup_close_cb (GtkWidget *widget, GdkEvent *event, gpointer data)
 {
   state *s = (state *) data;
   settings_cancel_cb (0, (gpointer) s);
+  return TRUE;
 }
 
 
